config/peerpods/podvm/lib.sh

```diff
@@ -205,11 +205,23 @@ function prepare_source_code() {
             error_exit "Failed to enable fips mode"
     fi
 
-    if [[ "$CONFIDENTIAL_COMPUTE_ENABLED" == "yes" ]]; then
-        sed 's/default SetPolicyRequest := true/default SetPolicyRequest := false/; s/default ExecProcessRequest := true/default ExecProcessRequest := false/' \
+    # links must be relative
+    if [[ "${AGENT_POLICY}" ]]; then
+        echo "Custom agent policy is being set through the AGENT_POLICY value"
+        echo ${AGENT_POLICY} | base64 -d > "${podvm_dir}"/files/etc/kata-opa/custom.rego
+        if [[ $? == 0 ]] && grep -q "agent_policy" "${podvm_dir}"/files/etc/kata-opa/custom.rego; then # checks policy validity
+            ln -sf custom.rego  "${podvm_dir}"/files/etc/kata-opa/default-policy.rego
+        else
+            error_exit "Invalid AGENT_POLICY value set, expected 64 based valid agent policy, got: \"${AGENT_POLICY}\""
+	fi
+    elif [[ "$CONFIDENTIAL_COMPUTE_ENABLED" == "yes" ]]; then
```

Comment:
We don't want to allow custom policy when CONFIDENTIAL_COMPUTE_ENABLED is yes?

Reply:
we don't? i think that if user override this explicitly we should accept it (it may be restricted than the default)

Reply:
Ok, custom agent policy gets priority, which is fair imho. Thanks